### PR TITLE
Add country to in-app marketplace search

### PR DESCRIPTION
**Pull request by kdevnel**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

The new React-powered marketplace can localise the results of searches from the WooCommerce.com API. In order to do this, it needs to supply the country code to the API. At the moment, it doesn't. This PR adds the necessary data from the store settings in order to regionalise search results.

Closes https://github.com/Automattic/woocommerce.com/issues/18031

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch locally and from `plugins/woocommerce-admin` run `pnpm run build`
2. Visit **WooCommerce > Extensions** from your development site (`wp-env`: http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions)
3. On the "Browse" tab choose a category from the category buttons
4. Take note of the API request in the network tab and the results.
5. Visit **WooCommerce > Settings > General** (`wp-env`: http://localhost:8888/wp-admin/admin.php?page=wc-settings
6. Change the Store Address > Country/State to something else (United States and South Africa are good examples)
7. Revisit the search and choose the same category.
8. Take note of the API request now including the new country code and that the results are different.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

Add country code to Marketplace search API requests.

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/admin/helper/class-wc-helper-admin.php [hunk 1]
```diff
@@ -39,11 +39,13 @@ public static function add_marketplace_settings( $settings ) {
 		$auth_user_data  = WC_Helper_Options::get( 'auth_user_data', array() );
 		$auth_user_email = isset( $auth_user_data['email'] ) ? $auth_user_data['email'] : '';
 
+		$store_location = wc_get_base_location();
 		$settings['wccomHelper'] = array(
 			'isConnected' => WC_Helper::is_site_connected(),
 			'connectURL'  => self::get_connection_url(),
 			'userEmail'   => $auth_user_email,
 			'userAvatar'  => get_avatar_url( $auth_user_email, array( 'size' => '48' ) ),
+			'storeCountry' => $store_location['country'],
```
Comment (raicem) [inline plugins/woocommerce/includes/admin/helper/class-wc-helper-admin.php]: Is it possible to inline `wc_get_base_location()` call? Not a big issue, just curious.

```suggestion
		$settings['wccomHelper'] = array(
			'isConnected' => WC_Helper::is_site_connected(),
			'connectURL'  => self::get_connection_url(),
			'userEmail'   => $auth_user_email,
			'userAvatar'  => get_avatar_url( $auth_user_email, array( 'size' => '48' ) ),
			'storeCountry' => wc_get_base_location()['country'],
```

**Review by raicem — APPROVED** [review]
Works as described. Thanks for fixing this.

File: plugins/woocommerce/includes/admin/helper/class-wc-helper-admin.php (hunk 1 above)
Comment (kdevnel) [reply plugins/woocommerce/includes/admin/helper/class-wc-helper-admin.php]: Yes, I prefer this. I tried it, but for some reason it seemed less clear to me. Looking at it now, that's far better.

**Review by andfinally — APPROVED** [review]
Thanks for looking into this, nice change!

It's interesting that WooCommerce Tax has a different product icon depending on whether the country is US or ZA. Maybe caching or something.

image

image

**Pull request merged**